python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -90,6 +95,28 @@
     "UnknownSerdesValue",
 ]
 
+_V = TypeVar("_V")
+
+
+class AssetKeyMap(Mapping["AssetKey", _V]):
```

Comment:
its not much of a lift to instead of just supporting `AssetKey` keys, support any serdes-able keys. 

I was thinking Instead of the to_string-ed key -> value approach currently used at [1], we could pack them in the style of the `items` API and do something like `{'__serdes_map_items__': [[k, v] for k, v in map.items()]}`

this idea applies even if we change the detection scheme